packages/documentation/plugins/remark/gatsby-remark-mdx-mermaid/index.js

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable */
+const visit = require('unist-util-visit');
+const remark = require('remark');
+const mermaid = require('./remark-mermaid');
+const vfile = require('to-vfile');
+
+module.exports = ({ markdownAST, markdownNode }, { language = 'mermaid', theme = 'default' } = {}) => {
+  visit(markdownAST, 'code', async (node) => {
+    const lang = (node.lang || '').toLowerCase();
+    const fileContent = vfile.readSync(markdownNode.fileAbsolutePath);
+
+    if (lang === language) {
+      await remark()
+        .use(mermaid)
+        .process(fileContent, (err, file) => {
+          if (err) throw err;
+
+          const mermaidSrcMatch = file.messages.filter(
```

Comment:
finds the specific mermaid code line that matches the current SVG. This handles the case where there are multiple mermaid graphs on a page.